WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationMessageTimerUpdateEventProcessor.swift

```diff
@@ -26,15 +27,51 @@ protocol ConversationMessageTimerUpdateEventProcessorProtocol {
     ///
     /// - Parameter event: A conversation message timer update event.
 
-    func processEvent(_ event: ConversationMessageTimerUpdateEvent) async throws
+    func processEvent(_ event: ConversationMessageTimerUpdateEvent) async
 
 }
 
 struct ConversationMessageTimerUpdateEventProcessor: ConversationMessageTimerUpdateEventProcessorProtocol {
 
-    func processEvent(_: ConversationMessageTimerUpdateEvent) async throws {
-        // TODO: [WPB-10171]
-        assertionFailure("not implemented yet")
+    let userRepository: any UserRepositoryProtocol
+    let conversationRepository: any ConversationRepositoryProtocol
+    let conversationLocalStore: any ConversationLocalStoreProtocol
+    let messageRepository: any MessageRepositoryProtocol
+
+    func processEvent(_ event: ConversationMessageTimerUpdateEvent) async {
+        let userID = event.senderID
+        let conversationID = event.conversationID
+        let timer = Double(event.newTimer ?? 0)
+        let timestamp = event.timestamp
+
+        let conversation = await conversationRepository.fetchOrCreateConversation(
+            id: conversationID.uuid,
+            domain: conversationID.domain
+        )
+
+        let timeoutValue = timer / 1000
+        let timeout: MessageDestructionTimeoutValue = .init(rawValue: timeoutValue)
+        let currentTimeout = await conversationLocalStore.conversationMessageDestructionTimeout(conversation)
+
+        if currentTimeout != timeout {
+
+            let messageType: MessageType = .messageTimerUpdate(
+                sender: (userID.uuid, userID.domain),
+                date: timestamp,
+                timeoutValue: timeoutValue
+            )
+
+            await messageRepository.addMessageToConversation(
```

Comment:
I would also expect adding a system message to be done on the local store, since it doesn't get sent via remote.

Reply:
fixed in https://github.com/wireapp/wire-ios/pull/2098/commits/36461566dd6fbe0e4de522b111e9ddb63e5d7214